newIDE/app/src/InstancesEditor/index.js

```diff
@@ -804,16 +811,72 @@ export default class InstancesEditor extends Component<Props> {
     this.viewPosition.scrollTo(x, y);
   }
 
-  centerView() {
+  zoomToFitContent() {
+    const instanceMeasurer = this.instancesRenderer.getInstanceMeasurer();
+    let contentAABB: ?Rectangle;
+    const getInstanceRectangle = new gd.InitialInstanceJSFunctor();
+    // $FlowFixMe - invoke is not writable
+    getInstanceRectangle.invoke = instancePtr => {
+      // $FlowFixMe - wrapPointer is not exposed
+      const instance = gd.wrapPointer(instancePtr, gd.InitialInstance);
+      if (!contentAABB) {
+        contentAABB = instanceMeasurer.getInstanceAABB(
+          instance,
+          new Rectangle()
+        );
+      } else {
+        contentAABB.union(
+          instanceMeasurer.getInstanceAABB(instance, new Rectangle())
+        );
+      }
+    };
+    const { initialInstances } = this.props;
+    if (initialInstances.getInstancesCount() > 0) {
+      // $FlowFixMe - JSFunctor is incompatible with Functor
+      initialInstances.iterateOverInstances(getInstanceRectangle);
+
+      if (contentAABB) {
+        const idealZoom = this.viewPosition.fitToRectangle(contentAABB);
+        this.setZoomFactor(idealZoom);
+      }
+    }
+  }
+
+  zoomToInitialPosition() {
     const x = this.props.project.getGameResolutionWidth() / 2;
     const y = this.props.project.getGameResolutionHeight() / 2;
     this.viewPosition.scrollTo(x, y);
+    this.setZoomFactor(1);
   }
 
-  centerViewOn(instances: Array<gdInitialInstance>) {
-    if (!instances.length) return;
+  zoomToFitSelection(instances: Array<gdInitialInstance>) {
+    if (instances.length === 0) return;
+    const [firstInstance, ...otherInstances] = instances;
```

Comment:
Is there a risk this throws an error if there is only 1 instance? (trying to decompose `otherInstances`)

Reply:
No, `otherInstances` will be `[]` in the case there is only one instance.